pinecone/index_connection.go

```diff
@@ -49,11 +56,88 @@ func newIndexConnection(in newIndexParameters) (*IndexConnection, error) {
 	return &idx, nil
 }
 
+// Close closes the connection to a Pinecone index.
+//
+// Returns a pointer to an IndexConnection object and an error if the connection cannot be closed.
```

Comment:
This is incorrect, the `Close()` only returns an `error` if there is one, otherwise it returns `nil`: https://github.com/pinecone-io/go-pinecone/blob/01670f82bd275b9be162c0373b03c5f25cd859e3/pinecone/index_connection.go#L54

I would also update this to explicitly refer to this closing the `grpc.ClientConn`.